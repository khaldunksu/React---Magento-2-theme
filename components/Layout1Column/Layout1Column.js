import React from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import Header from "../Header/Header"

export function Layout1Column(props) {
    const {children} = props;
    return (
            <div className="container js-hook__one-col-layout">

                <div className="layout-column1__header js-hook__one-col-layout-header">
                    <Header/>
                </div>
                <div className="layout-column1__content js-hook__one-col-layout-content"></div>
                <div className="layout-column1__footer js-hook__one-col-layout-footer"></div>
                {children}
            </div>
    );

}

Layout1Column.propTypes = {
    children: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
};
Layout1Column.defaultProps = {
    children: null,
};

export default connect(state => state)(Layout1Column);
