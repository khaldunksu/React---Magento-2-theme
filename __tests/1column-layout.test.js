import { shallow } from "enzyme";
import React from "react";
import Router from "next/router";
import toJson from "enzyme-to-json";
import { Layout1Column } from "../components/Layout1Column/Layout1Column";

const mockedRouter = { push: () => {} };
Router.router = mockedRouter;

describe("Component Page matches snapshot", () => {
    it("renders page layout correctly", () => {
        const wrapper = shallow(<Layout1Column />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

describe("Component Layout1Column", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Layout1Column />);
    });

    it("renders the dumb component", () => {
        expect(wrapper.length).toEqual(1);
    });

    it("contains main blocks", () => {
        expect(wrapper.find(".js-hook__one-col-layout-header").length).toBe(1);
        expect(wrapper.find(".js-hook__one-col-layout-content").length).toBe(1);
        expect(wrapper.find(".js-hook__one-col-layout-footer").length).toBe(1);
    });
});