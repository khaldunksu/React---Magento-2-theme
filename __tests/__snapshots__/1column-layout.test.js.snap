// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Page matches snapshot renders page layout correctly 1`] = `
<div
  className="container js-hook__one-col-layout"
>
  <div
    className="layout-column1__header js-hook__one-col-layout-header"
  />
  <div
    className="layout-column1__content js-hook__one-col-layout-content"
  />
  <div
    className="layout-column1__footer js-hook__one-col-layout-footer"
  />
</div>
`;
